tests/functional/raw-node/test/GCP/object/put.js

```diff
@@ -0,0 +1,116 @@
+const assert = require('assert');
+const { GCP } = require('../../../../../../lib/data/external/GCP');
+const { makeGcpRequest } = require('../../../utils/makeRequest');
+const { gcpRequestRetry } = require('../../../utils/gcpUtils');
+const { getRealAwsConfig } =
+    require('../../../../aws-node-sdk/test/support/awsConfig');
+
+const credentialOne = 'gcpbackend';
+const bucketName = `somebucket-${Date.now()}`;
+
+describe('GCP: PUT Object', function testSuite() {
+    this.timeout(30000);
+    const config = getRealAwsConfig(credentialOne);
+    const gcpClient = new GCP(config);
+
+    before(done => {
+        gcpRequestRetry({
+            method: 'PUT',
+            bucket: bucketName,
+            authCredentials: config.credentials,
+        }, 0, err => {
+            if (err) {
+                process.stdout.write(`err in creating bucket ${err}\n`);
+            }
+            return done(err);
+        });
+    });
+
+    after(done => {
+        gcpRequestRetry({
+            method: 'DELETE',
+            bucket: bucketName,
+            authCredentials: config.credentials,
+        }, 0, err => {
+            if (err) {
+                process.stdout.write(`err in deleting bucket ${err}\n`);
+            }
+            return done(err);
+        });
+    });
+
+    afterEach(function afterFn(done) {
+        makeGcpRequest({
+            method: 'DELETE',
+            bucket: bucketName,
+            objectKey: this.currentTest.key,
+            authCredentials: config.credentials,
+        }, err => {
+            if (err) {
+                process.stdout.write(`err in deleting object ${err}\n`);
+            }
+            return done(err);
+        });
+    });
+
+    describe('with existing object in bucket', () => {
+        beforeEach(function beforeFn(done) {
+            this.currentTest.key = `somekey-${Date.now()}`;
+            makeGcpRequest({
+                method: 'PUT',
+                bucket: this.currentTest.bucketName,
+                objectKey: this.currentTest.key,
+                authCredentials: config.credentials,
+            }, (err, res) => {
+                if (err) {
+                    process.stdout.write(`err in putting object ${err}\n`);
+                    return done(err);
+                }
+                this.currentTest.uploadId =
+                    res.headers['x-goog-generation'];
+                return done();
+            });
+        });
+
+        it('should overwrite object', function testFn(done) {
+            gcpClient.putObject({
+                Bucket: bucketName,
+                Key: this.test.key,
+            }, (err, res) => {
+                assert.notStrictEqual(res.VersionId, this.test.uploadId);
+                return done();
+            });
+        });
+    });
+
+    describe('without existing object in bucket', () => {
+        beforeEach(function beforeFn(done) {
```

Comment:
Looks like no need for `beforeEach` hook here.

Reply:
I have that beforeEach to set a variable used in an afterEach

Reply:
It's minor, but can you define `this.test.key` in the `it` block instead, before putting the object? Then we can remove the hook.

Reply:
yes. that is possible.